presto-parser/src/main/java/com/facebook/presto/sql/tree/Identifier.java

```diff
@@ -87,12 +87,12 @@ public boolean equals(Object o)
         }
 
         Identifier that = (Identifier) o;
-        return Objects.equals(value, that.value);
```

Comment:
Syntactically, yes, they are different.

The only reason the AST nodes implement equals and hashcode is so that we can make assertions during testing. The analyzer uses its own set of rules to decide whether things are equal or equivalent based on the semantics of the query and operations